Aplib.Extensions/Actions/PathfinderAction.cs

```diff
@@ -0,0 +1,33 @@
+using Aplib.Core;
+using Aplib.Core.Belief.BeliefSets;
+using Aplib.Core.Intent.Actions;
+
+namespace Aplib.Extensions.Actions
+{
+    public class PathfinderAction<TBeliefset, T> : Action<TBeliefset>
+        where TBeliefset : IBeliefSet
+    {
+        public PathfinderAction(IMetadata metadata, IPathfinder<T> pathfinder, System.Func<TBeliefset, T> getCurrentLocation, System.Func<TBeliefset, T> getEndLocation, System.Action<TBeliefset, T> effect)
+            : base(metadata, ExecutePathFinder(pathfinder, getCurrentLocation, getEndLocation, effect))
+        {
+        }
+
+        public PathfinderAction(IPathfinder<T> pathfinder, System.Func<TBeliefset, T> getCurrentLocation, System.Func<TBeliefset, T> getEndLocation, System.Action<TBeliefset, T> effect)
+            : this(null, pathfinder, getCurrentLocation, getEndLocation, effect)
```

Comment:
`null` should be replaced with `new Metadata()`, at least, that's what we do every where.